src/components/fields/submit-button/types.ts

```diff
@@ -3,4 +3,6 @@ import { IFrontendEngineBaseFieldJsonSchema, TComponentOmitProps } from "../../f
 
 export interface ISubmitButtonSchema
 	extends Omit<IFrontendEngineBaseFieldJsonSchema<"submit">, "validation">,
-		TComponentOmitProps<ButtonProps> {}
+		TComponentOmitProps<ButtonProps, "disabled"> {
+	disabled?: boolean | "invalid-form" | undefined;
```

Comment:
decided to add to the `disabled` flag instead of creating a new attribute because the latter option will create ambiguity in behaviour, e.g. should a button stay disabled if `disabled=false` and new attribute defines the button to be disabled if form is invalid?

Reply:
I'm personally ok with this, have seen similar options in the wild, where false means always enabled (or disabled) and true means always the opposite, with additional typings for custom behaviour